onnxruntime/core/providers/coreml/builders/impl/binary_op_builder.cc

```diff
@@ -131,9 +145,13 @@ Status BinaryOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const
     } else if (op_type == "Sub") {
       coreml_op_type = "sub";
     } else if (op_type == "Div") {
-      // we support fp32/fp16 currently. when we add support for integers we need to check the type and use
-      // "floor_div" or "real_div" accordingly
-      coreml_op_type = "real_div";
+      // Use "floor_div" op for integer division (int32 or int64)
+      // use "real_div" for float division (fp16 or fp32)
+      if (ShouldUseFloorDiv(node, logger)) {
+        coreml_op_type = "floor_div";
+      } else {
+        coreml_op_type = "real_div";
```

Comment:
So we have int64 unit tests for Div

Reply:
Ah to summarize offline conversation for visibility -- div int64 unit test exists [here](https://github.com/microsoft/onnxruntime/blob/15b3f086fae73702b62ba3da8a10e6ad4725efd8/onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc#L886) and covers the expected rounding behavior. This unit test passed previously (before the addition of floor_div handling) -- [CoreML docs](https://apple.github.io/coremltools/_modules/coremltools/converters/mil/mil/ops/defs/iOS15/elementwise_binary.html#real_div) don't specify expected behavior for real_div when using int32, but from the unit tests, it seems to have the same behavior as ONNX div. So using floor_div is optional (behavior-wise), but left the floor_div handling in because I thought it didn't hurt